docs/tutorials/data_provider.md

```diff
@@ -121,6 +145,7 @@ assert_eq!(provider.cache.lock().unwrap().len(), 0);
 
 assert_eq!(
     "こんにちは世界",
+    // Note: It is necessary to use `try_new_unstable` with LruDataCache.
```

Comment:
thought: this seems weird from client perpective. If we provided a cache we could actually make it implement `AnyProvider`

Reply:
I would love to hear how you think we could implement a cache implementing `AnyProvider` on top of an inner provider that implements `BufferProvider`.

Reply:
With the registry we can do it, no?

Reply:
Exactly; using the registry is the only way one _could_ do it. Maybe we don't align on whether the registry is a good solution; I think it's a solution that is generally quite bad and should be avoided if there is another viable option. In this case, we can put the cache into userland to circumvent the need for the registry.